lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -533,6 +555,20 @@ assignMemoryLayouts(scf::ForOp &forOp,
               getBlockedEncoding(loadOp, axisInfoAnalysis);
       }
     }
+
+    if (loadInfo.sharedEncoding && !isTMALoad) {
+      auto registerTy = cast<RankedTensorType>(op.getResultTypes()[0]);
+      auto vecBytes = getCopyVecBytes(registerTy, loadInfo.sharedEncoding);
+
+      if (vecBytes < 4) {
+        // At least 4 bytes need to be consecutive for cp.async
+        loadInfo.sharedEncoding = nullptr;
+        continue;
+      }
+    } else {
+      loadsToPipeline.insert(&op);
+    }
```

Comment:
can we consolidate it with the existing check:
https://github.com/triton-lang/triton/blob/main/lib/Dialect/TritonGPU/Transforms/Pipeliner/AssignLatencies.cpp#L86

Reply:
Done, this is a lot nicer